CMakeLists.txt

```diff
@@ -144,7 +144,7 @@ add_definitions( -D__STDC_LIMIT_MACROS )
 add_definitions( -D__STDC_CONSTANT_MACROS )
 add_definitions( -DCOMMON_CLANG_EXPORTS )
 if (NOT WIN32)
-    add_definitions( -DLIBCOMMON_CLANG_NAME="lib${COMMON_CLANG_LIBRARY_NAME}.so")
+    add_definitions(-DLIBCOMMON_CLANG_NAME="lib${COMMON_CLANG_LIBRARY_NAME}.so.${LLVM_VERSION_MAJOR}")
```

Comment:
I was noted that this might not work for all versions for LLVM. Looks like LLVM slightly steps out from usual practice to name libraries. For example, for 7.0.0 it creates (we reuse LLVM helper cmake functions to build):
* libopencl-clang.so.7
* libopencl-clang.so -> libopencl-clang.so.7

But for 7.1.0:
* libopencl-clang.so.7.1
* libopencl-clang.so -> libopencl-clang.so.7.1

And SONAME is programmed as `.so.7.1` inside binary instead of usual `.so.7`:
```
libopencl-clang.so.7.1 | grep SONAME
  SONAME               libopencl-clang.so.7.1
```

Hence, we need to avoid guessing SONAME and instead get it dynamically. Probably `$<TARGET_SONAME_FILE:${COMMON_CLANG_LIBRARY_NAME}>` construction might help.

Reply:
I just uploaded new version which should address this case. Indeed, $<TARGET_SONAME_FILE:${COMMON_CLANG_LIBRARY_NAME}> helps here.